tests/unit/containers/edit-profile/password-security-tab/PasswordSecurityTab.spec.jsx

```diff
@@ -1,18 +1,116 @@
 import {
   screen,
-  render,
   fireEvent,
+  waitFor,
   waitForElementToBeRemoved
 } from '@testing-library/react'
+import {
+  renderWithProviders,
+  mockAxiosClient,
+  TestSnackbar
+} from '~tests/test-utils'
 import PasswordSecurityTab from '~/containers/edit-profile/password-security-tab/PasswordSecurityTab'
+import { AuthService } from '~/services/auth-service'
+
+import { URLs } from '~/constants/request'
+import { expect } from 'vitest'
+
+const userDataMock = {
+  _id: 123456
+}
+
+vi.mock('~/services/auth-service', () => ({
+  AuthService: {
+    changePassword: vi.fn()
+  }
+}))
 
 const changeInputValue = (label, value) => {
   fireEvent.change(label, { target: { value } })
 }
 
 describe('PasswordSecurityTab', () => {
   beforeEach(() => {
-    render(<PasswordSecurityTab />)
+    renderWithProviders(
+      <TestSnackbar>
+        <PasswordSecurityTab user={userDataMock} />
+      </TestSnackbar>
+    )
+  })
+
+  it('should show Snackbar after positive response', async () => {
+    mockAxiosClient
+      .onPatch(`${URLs.auth.changePassword}/${userDataMock}`)
+      .reply(200)
+    const currentPasswordInput = screen.getByLabelText(
+      /editProfilePage.profile.passwordSecurityTab.currentPassword/i
+    )
+    const passwordInput = screen.getByLabelText(
+      /editProfilePage.profile.passwordSecurityTab.newPassword/i
+    )
+
+    const confirmPasswordInput = screen.getByLabelText(
+      /editProfilePage.profile.passwordSecurityTab.retypePassword/i
+    )
+
+    const saveButton = screen.getByText(
+      'editProfilePage.profile.passwordSecurityTab.savePassword'
+    )
+
+    fireEvent.change(currentPasswordInput, { target: { value: '12345qwert' } })
+    fireEvent.change(passwordInput, { target: { value: '12345qwertY' } })
+    fireEvent.change(confirmPasswordInput, { target: { value: '12345qwertY' } })
+
+    await waitFor(() => {
+      fireEvent.click(saveButton)
+    })
+
+    await waitFor(() => {
+      expect(AuthService.changePassword).toHaveBeenCalledWith(
+        userDataMock._id,
+        {
+          password: '12345qwertY',
+          currentPassword: '12345qwert'
+        }
+      )
+    })
+  })
+
+  it('should do not save empty fields', async () => {
+    const currentPasswordInput = screen.getByLabelText(
+      /editProfilePage.profile.passwordSecurityTab.currentPassword/i
+    )
+    const passwordInput = screen.getByLabelText(
+      /editProfilePage.profile.passwordSecurityTab.newPassword/i
+    )
+    const confirmPasswordInput = screen.getByLabelText(
+      /editProfilePage.profile.passwordSecurityTab.retypePassword/i
+    )
+    const saveButton = screen.getByText(
+      /editProfilePage.profile.passwordSecurityTab.savePassword/i
+    )
```

Comment:
Please move these elements into setup function for reusability as following:

```jsx
const setupChangePasswordFields = () => {
     const currentPasswordInput = screen.getByLabelText(
      /editProfilePage.profile.passwordSecurityTab.currentPassword/i
    )
    const passwordInput = screen.getByLabelText(
      /editProfilePage.profile.passwordSecurityTab.newPassword/i
    )
    const confirmPasswordInput = screen.getByLabelText(
      /editProfilePage.profile.passwordSecurityTab.retypePassword/i
    )
    const saveButton = screen.getByText(
      /editProfilePage.profile.passwordSecurityTab.savePassword/i
    )
    
    return { currentPasswordInput, passwordInput, confirmPasswordInput, saveButton }
}
```

Reply:
Moved! Thanks